packages/playground/src/weblets/freeflow.vue

```diff
@@ -210,10 +223,22 @@ async function deploy() {
 function updateSSHkeyEnv(selectedKeys: string) {
   selectedSSHKeys.value = selectedKeys;
 }
+
+watch(
+  () => selectionDetails.value?.domain?.enabledCustomDomain,
```

Comment:
can we move this logic to the network tab?
instead of passing readonly flag, i suggest to pass the customDomainEnabled and check for both of customdomain enabled and ipv4 values then handle the wireguard option

Reply:
I have an open issue for enhancing the entire component, please if you want to do this enhancement in this PR, check with @AhmedHanafy725

https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3294

Reply:
It's also a valid comment tbh, we can do it here to avoid code redundancy

Reply:
- Established a new constant by computing the options for IPv4 and custom domains to decide whether the Wireguard option should be read-only or not.
- Monitored the 'isWireguardReadonly' constant to determine the value of the Wireguard feature based on it.